src/sentry/api/endpoints/organization_plugins_configs.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+import six
+
+from sentry.plugins.base import plugins
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.plugin import PluginSerializer
+from sentry.models import ProjectOption
+
+
+class OrganizationPluginsConfigsEndpoint(OrganizationEndpoint):
+    def get(self, request, organization):
+        all_plugins = dict([(p.slug, p) for p in plugins.all()])
+
+        # find all the keys that will tell us if a plugin is enabled OR configured
+        keys_to_check = []
+        for slug, plugin in six.iteritems(all_plugins):
+            keys_to_check.append("%s:enabled" % slug)
+            if plugin.required_field:
```

Comment:
where is the `required_field` getting set again?

Reply:
@MeredithAnya They are defined in the `plugin.py` files where we sub-class the plugin. Ex:
```
class TrelloPlugin(CorePluginMixin, IssuePlugin2):
    description = "Create Trello cards on issues"
    slug = "trello"
    title = "Trello"
    conf_title = title
    conf_key = "trello"
    auth_provider = None
    resource_links = [("Trello Setup Instructions", SETUP_URL)] + CorePluginMixin.resource_links
    required_field = "key"

```

Reply:
ahh okays cool - I haven't pulled from master in a while so I couldn't find when searching in my editor lol

Reply:
@MeredithAnya Hahah yea, that change was merged today :)